src/plugins/telemetryTable/collections/TableRowCollection.js

```diff
@@ -0,0 +1,482 @@
+                const keyString = this.openmct.objects.makeKeyString(telemetryObject.identifier);
+                let requestOptions = this.buildOptionsFromConfiguration(telemetryObject);
+                let columnMap = this.getColumnMapForObject(keyString);
+                let limitEvaluator = this.openmct.telemetry.limitEvaluator(telemetryObject);
+
+                this.incrementOutstandingRequests();
+
+                const telemetryProcessor = this.getTelemetryProcessor(keyString, columnMap, limitEvaluator);
+                const telemetryRemover = this.getTelemetryRemover();
+
+                this.removeTelemetryCollection(keyString);
+
+                this.telemetryCollections[keyString] = this.openmct.telemetry
+                    .requestTelemetryCollection(telemetryObject, requestOptions);
+
+                this.telemetryCollections[keyString].on('remove', telemetryRemover);
+                this.telemetryCollections[keyString].on('add', telemetryProcessor);
+                this.telemetryCollections[keyString].load();
+
+                this.decrementOutstandingRequests();
+
+                this.telemetryObjects[keyString] = {
+                    telemetryObject,
+                    keyString,
+                    requestOptions,
+                    columnMap,
+                    limitEvaluator
+                };
+            }
+
+            removeObject(objectIdentifier) {
+                const keyString = this.openmct.objects.makeKeyString(objectIdentifier);
+                let removed = [];
+
+                this.rows = this.rows.filter((row) => {
+                    if (row.objectKeyString !== keyString) {
+                        removed.push(row);
+
+                        return false;
+                    } else {
+                        return true;
+                    }
+                });
+
+                this.emit('remove', removed);
+                this.removeTelemetryCollection(keyString);
+                delete this.telemetryObjects[keyString];
+            }
+
+            removeTelemetryCollection(keyString) {
+                if (this.telemetryCollections[keyString]) {
+                    this.telemetryCollections[keyString].destroy();
+                    this.telemetryCollections[keyString] = undefined;
+                    delete this.telemetryCollections[keyString];
+                }
+            }
+
+            getTelemetryProcessor(keyString, columnMap, limitEvaluator) {
+                return (telemetry) => {
+                    //Check that telemetry object has not been removed since telemetry was requested.
+                    if (!this.telemetryObjects[keyString]) {
+                        return;
+                    }
+
+                    // only cache realtime
+                    if (this.paused) {
+                        this.processFunctions.push(this.processTelemetryData.bind(this, telemetry, columnMap, keyString, limitEvaluator));
```

Comment:
Let's name this something like `pausedProcessFunctions` or something? I first saw this array referenced outside of this function and it took me a minute to figure out what it was.

Reply:
updated to delayedActions